src/D2L.Bmx/OktaAuthenticator.cs

```diff
@@ -130,6 +136,114 @@ private bool TryAuthenticateFromCache(
 		return true;
 	}
 
+	private async Task<IOktaAuthenticatedClient?> TryAuthenticateWithDSSOAsync(
+		string org,
+		string user,
+		IOktaClientFactory oktaClientFactory,
+		bool experimental
+	) {
+		await using IBrowser? browser = await Browser.LaunchBrowserAsync( experimental );
+		if( browser is null ) {
+			return null;
+		}
+
+		Console.WriteLine( "Attempting to automatically login using DSSO." );
+		var cancellationTokenSource = new CancellationTokenSource( TimeSpan.FromSeconds( 10 ) );
+		var sessionIdTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		var userEmailTaskProducer = new TaskCompletionSource<string?>( TaskCreationOptions.RunContinuationsAsynchronously );
+		string? sessionId;
+		string? userEmail;
+
+		try {
+			var page = await browser.NewPageAsync().WaitAsync( cancellationTokenSource.Token );
+			string baseAddress = $"https://{org}.okta.com/";
+			int attempt = 1;
+
+			page.Load += ( _, _ ) => _ = GetSessionCookieAsync();
+			page.Response += ( _, responseCreatedEventArgs ) => _ = GetOktaUserEmailAsync(
+				responseCreatedEventArgs.Response
+			);
+			await page.GoToAsync( baseAddress );
+			sessionId = await sessionIdTaskProducer.Task.WaitAsync( cancellationTokenSource.Token );
+			userEmail = await userEmailTaskProducer.Task.WaitAsync( cancellationTokenSource.Token );
+
+			async Task GetSessionCookieAsync() {
+				var url = new Uri( page.Url );
+				if( url.Host == $"{org}.okta.com" ) {
+					string title = await page.GetTitleAsync();
+					if( title.Contains( "sign in", StringComparison.OrdinalIgnoreCase ) ) {
+						if( attempt < 3 && url.AbsolutePath == "/" ) {
```

Comment:
`==` is different than the poc `!=`. I found that for our sign in page at least that the absolute path was `/`